packages/amplify-category-auth/src/provider-utils/awscloudformation/import/index.ts

```diff
@@ -1232,36 +1237,36 @@ export const headlessImport = async (
   const answers: ImportAnswers = {
     authSelections: resourceParameters.authSelections,
     resourceName: resourceParameters.resourceName,
-    userPoolId: currentEnvSpecificParameters.userPoolId,
+    userPoolId: resolvedEnvParams.userPoolId,
   };
 
   try {
-    answers.userPool = await cognito.getUserPoolDetails(currentEnvSpecificParameters.userPoolId);
+    answers.userPool = await cognito.getUserPoolDetails(resolvedEnvParams.userPoolId);
   } catch (error) {
     if (error.name === 'ResourceNotFoundException') {
-      throw new Error(importMessages.UserPoolNotFound(currentEnvSpecificParameters.userPoolName, currentEnvSpecificParameters.userPoolId));
+      throw new Error(importMessages.UserPoolNotFound(resolvedEnvParams.userPoolName, resolvedEnvParams.userPoolId));
     }
 
     throw error;
   }
 
-  const validationResult = await validateUserPool(cognito, identity, questionParameters, answers, currentEnvSpecificParameters.userPoolId);
+  const validationResult = await validateUserPool(cognito, identity, questionParameters, answers, resolvedEnvParams.userPoolId);
 
   if (typeof validationResult === 'string') {
     throw new Error(validationResult);
   }
 
   // Get app clients based on passed in previous values
-  answers.appClientWeb = questionParameters.webClients!.find((c) => c.ClientId! === currentEnvSpecificParameters.webClientId);
+  answers.appClientWeb = questionParameters.webClients!.find((c) => c.ClientId! === resolvedEnvParams.webClientId);
```

Comment:
replace ! with ? here. `.find` already returns a nullable value, so this assertion can be removed.